corehq/apps/reports/views.py

```diff
@@ -1104,6 +1120,22 @@ def _render_report_configs(request, configs, domain, owner_id, couch_user, email
     return response.content.decode("utf-8"), excel_attachments
 
 
+def _is_location_restricted(domain, config, couch_user):
+    '''
+    Checks if request is location restricted for the asked report.
+    Returns True if request is restricted for the asked user and the report. Returns False otherwise.
+    '''
+    if not toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(domain):
+        return False
+    # Check if report is location safe
+    if any(["computed_owner_location" in report_filter for report_filter in config.filters.keys()]):
```

Comment:
What is this doing?  Here's how reports are checked for safety [in the UI](https://github.com/dimagi/commcare-hq/blob/8c6cddad91b2b8d241214a6ab30fa7e556d655cc/corehq/apps/userreports/reports/view.py#L140-L145):
https://github.com/dimagi/commcare-hq/blob/8c6cddad91b2b8d241214a6ab30fa7e556d655cc/corehq/apps/userreports/reports/util.py#L21-L32

Reply:
It checks if any of the report filter is `Case Owner (Location)` which ensures that a report is location safe ([Ref.](https://confluence.dimagi.com/display/saas/%5BTo+Archive%5D+Enable+creation+of+report+builder+reports+that+are+location+safe)).

Reply:
Is it possible to find the `ReportConfiguration` from the `ReportConfig`?  I do think that both places need to do the same check (ideally reusing the logic).  The code snippet I shared above is the canonical definition of what a location-safe UCR is - one with a filter with a choice provider that's specifically marked as [location_safe](https://github.com/dimagi/commcare-hq/blob/2be6c2f001712852433cd985fe85a470eb94987c/corehq/apps/userreports/reports/filters/choice_providers.py#L263-L265) or where [`location_filter = True`](https://github.com/dimagi/commcare-hq/blob/2be6c2f001712852433cd985fe85a470eb94987c/corehq/apps/reports_core/filters.py#L418-L420).

For UCRs built using report builder, filters with [any of these three](https://github.com/dimagi/commcare-hq/blob/2be6c2f001712852433cd985fe85a470eb94987c/corehq/apps/userreports/reports/builder/forms.py#L251-L270) formats will use location-safe choice providers.  I think you're correct that all three of those will contain the string `computed_owner_location`, assuming that something turns the `/` into a `_`.  However, we have no such guarantees with UCRs made outside report builder, and it also seems more future-proof to check the location safety of the filters directly rather than inferring location safety based on a pattern seen in the name.